Utils/Editor/FindUnusedAssetsWindow.cs

```diff
@@ -0,0 +1,241 @@
+
+		[MenuItem("DUCK/Find Unused Assets")]
+		private static void Init()
+		{
+			window = (FindUnusedAssetsWindow) GetWindow(typeof(FindUnusedAssetsWindow));
+			window.Show();
+			window.maxSize = new Vector2(WIDTH, 1080);
+			window.minSize = new Vector2(100, WIDTH);
+			window.titleContent = new GUIContent("Find Unused Assets");
+
+			hasEditorLagBeenFound = true;
+		}
+
+		private static void Search()
+		{
+			List<string> usedAssets = FindUsedAssets();
+
+			if (usedAssets != null)
+			{
+				unusedAssets = FindUnusedObjects(FindAllProjectAssets(), usedAssets).OrderBy(a => a.Obj != null).ToList();
+				hasEditorLagBeenFound = true;
+			}
+			else
+			{
+				hasEditorLagBeenFound = false;
+			}
+		}
+
+		private void OnGUI()
+		{
+			EditorGUILayout.HelpBox("Does not include Scenes or Meta data.", MessageType.Info);
+			GUILayout.BeginHorizontal();
+			includeDuck = GUILayout.Toggle(includeDuck, "Include DUCK");
+			includePlugins = GUILayout.Toggle(includePlugins, "Include Plugins");
+			includeVendor = GUILayout.Toggle(includeVendor, "Include Vendor");
+			if (GUILayout.Button("Search"))
+			{
+				Search();
+			}
+			GUILayout.EndHorizontal();
+
+			if (!hasEditorLagBeenFound)
+			{
+				EditorGUILayout.HelpBox("Editor log not found! - You need to build your project before doing this!",
+					MessageType.Warning);
+			}
+			else if (unusedAssets != null)
+			{
+				GUILayout.Label(string.Format("Found {0} unused assets!", unusedAssets.Count), EditorStyles.boldLabel);
+				var scrollViewHeight = EditorGUIUtility.singleLineHeight * unusedAssets.Count;
+				var scrollViewPositionY = EditorGUIUtility.singleLineHeight * 6;
+				var scrollViewPosition = new Rect(0, scrollViewPositionY, WIDTH, window.position.height - scrollViewPositionY);
+				var scrollView = new Rect(0, 0, WIDTH, scrollViewHeight);
+				scrollPosition = GUI.BeginScrollView(scrollViewPosition, scrollPosition, scrollView, false, true);
+				for (var i = 0; i < unusedAssets.Count; i++)
+				{
+					var asset = unusedAssets[i];
+					if (asset != null)
+					{
+						var positionY = EditorGUIUtility.singleLineHeight * i;
+						GUILayout.BeginHorizontal();
+						if (asset.Obj != null)
+						{
+							GUI.Label(new Rect(10, positionY, 250, EditorGUIUtility.singleLineHeight), asset.Name);
+							if (GUI.Button(new Rect(260, positionY, 70, EditorGUIUtility.singleLineHeight), "Select"))
+							{
+								Selection.activeObject = asset.Obj;
+							}
+							if (GUI.Button(new Rect(330, positionY, 70, EditorGUIUtility.singleLineHeight), "Delete"))
+							{
+								AssetDatabase.DeleteAsset(asset.Path);
+							}
+						}
+						else
+						{
+							GUI.Label(new Rect(10, positionY, 250, EditorGUIUtility.singleLineHeight), asset.Path);
+						}
+						GUILayout.EndHorizontal();
+					}
+				}
+				GUI.EndScrollView();
+			}
+		}
+
+		private static IEnumerable<Asset> FindUnusedObjects(IEnumerable<string> assetList, ICollection<string> usedAssets)
```

Comment:
Would it be better to use ToArray instead of ToList in this instance, as the array is immutable. From what I can tell you want to ensure that the underlying collection is protected from modification, this is why you return IEnumerable. Given this is the case, creating an array has slightly less overhead than a list and is guaranteed to be immutable.